uiowa/uiowa
# Feature:  Allow site editors the ability to unset the default banner heading size

As a site owner, I can adjust my banner title size by using different heading levels so that there's more than one set size.

## Work completed in this pull request

<details>

- [x] Added "Headline Large, Medium, and Small" options in "Headline Sizes" group
- [x] Made the default option "Headline Large" if adding a new banner
- [x] Change "size" label to "heading" on banner title field
- [x] Move "Headline Sizes" up to be the first item in the Display Options
- [x] Update hook to set "Headline Large" as default for existing banners 
- [x] Remove legacy `bold-` classes from banner layout builder styles.
- [x] Adjusted small, medium, and large headline sizes to be smaller if banners are placed within `.layout--threecol, .layout--fourcol, .layout--twocol--25-75 .layout__region--first, .layout--twocol--75-25 .layout__region--second, .layout--twocol--33-67 .layout__region--first, .layout--twocol--67-33`
- [x] Added `banner--small` modifier padding-top percentage only for use on featured image banners.
</details>

## Future Slices
- Move "Headline Sizes" field under banner title and into a fieldset with help text below explaining how it all works. 

# How to test

Test on https://diversity.local.drupal.uiowa.edu/ and any other banner heavy V3 sites you can think of. 

```
rm -rf ./node_modules
yarn cache clean
yarn install
yarn workspace uids_base gulp --development
blt ds --site=sandbox.uiowa.edu
blt ds --site=diversity.uiowa.edu
drush @sandbox.local cr
drush @diversity.local cr
```

**Testing steps:**

1. Verify that `.headline--large` class is set as default on existing banners via update hook. 
2. Check the home page banner on sandbox and verify that the unsaved change in temp storage was updated. 
3. Verify that `Headine Large` is the default option when creating a new banner. 
4. Verify that there are no `bold-headline` classes being applied in the banner markup or in the layout builder headline styles. 
5. Play around with small, medium, large, and -none- options. 

**User stories:**

As a site owner, I can adjust my banner title size by using small, medium, large headline options so that there's more than one set size available.
